xclim/ensembles/_reduce.py

```diff
@@ -27,6 +28,85 @@
     MPL_INSTALLED = False
 
 
+def make_criteria(ds: xarray.Dataset | xarray.DataArray):
+    """Reshapes the input into a criteria 2D DataArray.
+
+    The reshaping preserves the "realization" dimension but stacks all other
+    dimensions and variables into a new "criteria" dimension, as expected by
+    functions :py:func:`xclim.ensembles._reduce.kkz_reduce_ensemble`
+    and :py:func:`xclim.ensembles._reduce.kmeans_reduce_ensemble`.
+
+    Parameters
+    ----------
+    ds : Dataset or DataArray
+        Must at least have a "realization" dimension. All values are considered independent "criterion" for the ensemble reduction.
+        If a Dataset, variables may have different sizes, but all must include the "realization" dimension.
+
+    Returns
+    -------
+    crit : DataArray
+        Same data, reshaped. Old coordinates (and variables) are available as a multiindex.
+
+    Notes
+    -----
+    One can get back to the original dataset with
+
+    >>> crit = make_criteria(ds)
+    >>> ds2 = crit.unstack("criteria").to_dataset("variables")
```

Comment:
The reason for build failures is that this is actually running! If we want, you can replace `ds` with one of the existing dataset aliases: https://github.com/Ouranosinc/xclim/blob/78ff493ded25907c4e17f0695950d1e04ff88d28/xclim/testing/tests/data.py#L108

Reply:
Dang, I thought that placing it in `notes` (and not `examples`) would make it _not_ run.

Reply:
All "datasets" in the namespace are actually DataArrays... I changed to a code-block instead. (this shouldn't run, right?)

Reply:
Correct!